openedx/features/colaraz_features/auth_pipeline.py

```diff
@@ -1,12 +1,26 @@
 """
 Authenitcation and Social Auth Pipeline methods for Colaraz's customizations
 """
+from openedx.features.colaraz_features.models import ColarazUserProfile
+
 
 def store_id_token(request, response, user=None, *args, **kwargs):
     """
     This method is used in SOCIAL_AUTH_PIPELINE. It stores 'id_token' from the User's
     data sent by Auth Provider to request's 'session' object.
     """
-    
     if user and response.has_key('id_token'):
         request.session['id_token'] = response['id_token']
+
+
+def update_colaraz_profile(request, response, user=None, *args, **kargs):
+    """
+    Updates ColarazUserProfile using data sent from Colaraz Auth Provider
+    """
+    if user:
+        instance, _ = ColarazUserProfile.objects.update_or_create(user=user, defaults={
+            'site_identifier': str(response.get('companyInfo', {}).get('url', '')).lower(),
```

Comment:
code will breaks if '**companyInfo**' key is there in response,  with value None i.e. response= {'companyInfo': None}.

Reply:
Hmmm, I don't believe it ever gonna happen, but still, let me see what I can do.

Reply:
Why do we need to explicitly convert the value into `str` here ?

Reply:
@danialmalik, The data sent by auth-provider is usually in unicode, so have to convert that into str to use lower() method on it.

Reply:
@tehreem-sadat, I have applied a try-except, couldn't think if any specific exception let me know if you have any in mind.

Reply:
instead of that, we can also go with this `str((response.get('companyInfo') or {}).get('url', '')).lower()` but I guess try catch is better solution from readablity point of view.